src/control/system/database_test.go

```diff
@@ -0,0 +1,287 @@
+//
+// (C) Copyright 2020 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package system
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net"
+	"testing"
+
+	"github.com/daos-stack/daos/src/control/build"
+	"github.com/daos-stack/daos/src/control/common"
+	"github.com/daos-stack/daos/src/control/logging"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/google/uuid"
+	"github.com/hashicorp/raft"
+	"github.com/pkg/errors"
+)
+
+func setupTestDatabase(t *testing.T, log logging.Logger, replicas []string) (*Database, func()) {
+	testDir, cleanup := common.CreateTestDir(t)
+
+	db := NewDatabase(log, &DatabaseConfig{
+		RaftDir:  testDir,
+		Replicas: replicas,
+	})
+	return db, cleanup
+}
+
+/*func TestSystem_DatabaseSnapshot(t *testing.T) {
```

Comment:
think we should have TODO notices for code that needs un-commenting later to ensure it gets tracked

Reply:
Yeah, all the commented-out tests will be uncommented before landing. That one in particular is already working again.